de_web_plugin.cpp

```diff
@@ -9013,6 +9020,28 @@ void DeRestPluginPrivate::updateSensorNode(const deCONZ::NodeEvent &event)
                                     updated = true;
                                 }
                             }
+                            else if (ia->id() == 0x050f) // Apparent power
+                            {
+                                if (updateType != NodeValue::UpdateInvalid)
+                                {
+                                    i->setZclValue(updateType, event.endpoint(), event.clusterId(), ia->id(), ia->numericValue());
+                                    pushZclValueDb(event.node()->address().ext(), event.endpoint(), event.clusterId(), ia->id(), ia->numericValue().u16);
+                                }
+
+                                quint16 power = ia->numericValue().u16;
+                                ResourceItem *item = i->item(RStatePower);
+
+                                if (item && power != -32768)
+                                {
+                                    if (i->modelId() == QLatin1String("TH1300ZB")) // Sinope thermostat
+                                    {
+                                        power = static_cast<quint16>(round((double)power / 1000.0)); // -> W
```

Comment:
Since power is an `uint16`, shouldn't a `power /= 1000` work as well?

Reply:
Work: yes, correct rounding: no ;)